typeid/typeid-go/typeid.go

```diff
@@ -10,44 +10,49 @@ import (
 )
 
 // TypeID is a unique identifier with a given type as defined by the TypeID spec
-type TypeID struct {
-	prefix string
+type TypeID[P Prefix] struct {
+	prefix P
```

Comment:
This structure wont be compatible with an interface where `New` returns type `T`.

Specifically, this change would make the approach in this [PR](https://github.com/jetpack-io/opensource/pull/185) undoable. You could not be able to do:

`id, err := typeid.New[MyID]()`

Reply:
Additionally, I just don't understand the value of this. You can still templatize the struct without turning the prefix into a generic type which makes the logic everywhere more complex.

Reply:
Mentioned it in the other response, but I'll change the prefix by to a `string`, and I think that's the only issue. I believe `TypeID` itself can be compatible with the `Prefix` interface, and by extension, compatible with your PR. Tell me if I'm misunderstanding something though.